src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs

```diff
@@ -257,8 +258,331 @@ private struct IntPtrAndHandle
+                ? Type.GetTypeFromHandle(typeHandle)
+                : null;
+
+            MethodBase? mb = (typePtr != IntPtr.Zero)
+                ? MethodBase.GetMethodFromHandle(methodHandle, typeHandle)
+                : MethodBase.GetMethodFromHandle(methodHandle);
+            if (mb == null)
+                return null;
+
+            var result = "{ " +
+                $"\"result\": {MakeMarshalTypeRecord((mb as MethodInfo)?.ReturnType ?? typeof(void))}, " +
+                $"\"typePtr\": {typePtr}, \"methodPtr\": {methodPtr}, " +
+                "\"parameters\": [";
+
+            int i = 0;
+            foreach (var p in mb.GetParameters()) {
+                if (i > 0)
+                    result += ", ";
+                result += MakeMarshalTypeRecord(p.ParameterType);
+                i++;
+            }
+
+            result += "] }";
+
+            return result;
+        }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2070:UnrecognizedReflectionPattern",
+            Justification = "Trimming doesn't affect types eligible for marshalling. Different exception for invalid inputs doesn't matter.")]
+        private static unsafe string GetAndEscapeStringProperty (Type type, string name) {
+            var info = type.GetProperty(
+                name, BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic
+            );
+
+            var value = info?.GetValue(null) as string;
+            if (value == null)
+                return "null";
+
+            var result = "\"";
+            foreach (var ch in value) {
+                switch (ch) {
+                    case '\'':
+                        result += '\'';
+                        continue;
+                    case '"':
+                        result += '\"';
+                        continue;
+                    case '\\':
+                        result += "\\\\";
+                        continue;
+                    case '\n':
+                        result += "\\n";
+                        continue;
+                }
+
+                if (ch < ' ') {
+                    result += "\\x";
+                    if (ch <= 0xF)
+                        result += '0';
+                    result += ((int)ch).ToString("X");
+                } else {
+                    result += ch;
+                }
+            }
+            result += '"';
+
+            return result;
+        }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2070:UnrecognizedReflectionPattern",
+            Justification = "Trimming doesn't affect types eligible for marshalling. Different exception for invalid inputs doesn't matter.")]
+        private static unsafe IntPtr GetMarshalMethodPointer (Type type, string name, out Type? returnType, out Type? parameterType) {
+            var info = type.GetMethod(
+                name, BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic
```

Comment:
I think this will scan whole type hierarchy, is that intentional?

Reply:
No, good point. It used to be intentional when all of these were instance methods, but they're not anymore

Reply:
This is a nonrecursive search since it's static and not flattenhierarchy, so this should be fine I think